parse/parser.go

```diff
@@ -200,16 +202,44 @@ func makePrefixExpression(buf TokenBuffer) (ast.Expression, []error) {
 	return exp, errs
 }
 
-// TODO: implement me w/ test cases :-)
-// MakeInfixExpression retrieves infix parse function from map
-// then parse expression with that function if exist.
 func makeInfixExpression(buf TokenBuffer, exp ast.Expression, pre precedence) (ast.Expression, []error) {
-	return nil, nil
+	errs := make([]error, 0)
+	expression := exp
+	curTok := buf.Peek(CURRENT)
+
+	for pre < nextPrecedence(buf) {
+		fn := infixParseFnMap[buf.Peek(NEXT).Type]
+
+		if fn == nil {
+			errs = append(errs, parseError{curTok.Type, "infix parse function not defined"})
+			return nil, errs
+		}
+		buf.Read()
+
+		var err []error
+		expression, err = fn(buf, expression)
+
+		if len(err) > 0 {
+			return nil, append(errs, err...)
+		}
+	}
+	return expression, nil
 }
 
-// TODO: implement me w/ test cases :-)
 func parseInfixExpression(buf TokenBuffer, exp ast.Expression) (ast.Expression, []error) {
-	return nil, nil
+	expression := &ast.InfixExpression{Left: exp,
```

Comment:
I think there's no test case for this function

Reply:
When we are executing this function, we can fully assume that the current token is **already** infix operator 

so before making `expression` we can `buf.Read()` and retrieve token.

Reply:
Indentation quite unreadable... how about
```
expression := &ast.InfixExpression{
    Left: ...
}